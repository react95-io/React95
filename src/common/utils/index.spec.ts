import {
  clamp,
  mapFromWindowsTheme,
  getDecimalPrecision,
  roundValueToStep,
  defaultTrue
} from './index';

describe('clamp', () => {
  it('should return passed value if its between min and max', () => {
    expect(clamp(4, 3, 5)).toBe(4);
  });

  it('should return min if value is smaller than min', () => {
    expect(clamp(2, 3, 5)).toBe(3);
  });

  it('should return max if value is bigger than max', () => {
    expect(clamp(6, 3, 5)).toBe(5);
  });

  it('should accept null as min', () => {
    expect(clamp(6, null, 5)).toBe(5);
  });

  it('should accept null as max', () => {
    expect(clamp(1, 2, null)).toBe(2);
  });
});

describe(defaultTrue, () => {
  it.each([
    [null, true],
    [undefined, true],
    [true, true],
    [false, false]
  ])('when %s, returns %s', (value, expected) => {
    expect(defaultTrue(value)).toEqual(expected);
  });
});

describe('mapFromWindowsTheme', () => {
  const theme = {
    ButtonAlternateFace: '#000000',
    ButtonDkShadow: '#000001',
    ButtonFace: '#000002',
    ButtonHilight: '#000003',
    ButtonLight: '#000004',
    ButtonShadow: '#000005',
    ButtonText: '#000006',
    ActiveBorder: '#000007',
    AppWorkspace: '#000008',
    Background: '#000009',
    InactiveBorder: '#00000a',
    Scrollbar: '#00000b',
    Window: '#00000c',
    WindowFrame: '#00000d',
    WindowText: '#00000e',
    ActiveTitle: '#00000f',
    GradientActiveTitle: '#000010',
    GradientInactiveTitle: '#000011',
    InactiveTitle: '#000012',
    InactiveTitleText: '#000013',
    TitleText: '#000014',
    Menu: '#000015',
    MenuBar: '#000016',
    MenuHilight: '#000017',
    MenuText: '#000018',
    GrayText: '#000019',
    Hilight: '#00001a',
    HilightText: '#00001b',
    HotTrackingColor: '#00001c',
    InfoText: '#00001d',
    InfoWindow: '#00001e'
  };

  it('should map corresponding properties directly', () => {
    expect(mapFromWindowsTheme('theme', theme)).toMatchSnapshot();
  });

  it('should support gradients and shadows and scale', () => {
    expect(
      mapFromWindowsTheme('theme', theme, {
        useGradients: true,
        useShadow: false,
        scale: 1.5
      })
    ).toMatchSnapshot();
  });
});

describe('getDecimalPrecision', () => {
  it('should return 0 when passed a round number', () => {
    expect(getDecimalPrecision(4)).toBe(0);
  });

  it('should return correct decimal precision', () => {
    expect(getDecimalPrecision(1.233)).toBe(3);
  });
});

describe('roundValueToStep', () => {
  it('should be able to round down', () => {
    expect(roundValueToStep(4, 3, 0)).toBe(3);
  });

  it('should be able to round up', () => {
    expect(roundValueToStep(5, 3, 0)).toBe(6);
  });
});
