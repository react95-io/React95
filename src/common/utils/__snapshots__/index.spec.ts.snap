// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapFromWindowsTheme should map corresponding properties directly 1`] = `
Object {
  "anchor": "#00001c",
  "anchorVisited": "#00001c",
  "borderDark": "#000005",
  "borderDarkest": "#000001",
  "borderLight": "#000004",
  "borderLightest": "#000003",
  "canvas": "#00000c",
  "canvasText": "#00000e",
  "canvasTextDisabled": "#000005",
  "canvasTextDisabledShadow": "#000003",
  "canvasTextInvert": "#00001b",
  "checkmark": "#00000e",
  "checkmarkDisabled": "#000019",
  "desktopBackground": "#000009",
  "flatDark": "#000005",
  "flatLight": "#000004",
  "focusSecondary": "#000003",
  "headerBackground": "#00000f",
  "headerNotActiveBackground": "#000012",
  "headerNotActiveText": "#000013",
  "headerText": "#000014",
  "hoverBackground": "#00001a",
  "material": "#000002",
  "materialDark": "#000012",
  "materialText": "#000006",
  "materialTextDisabled": "#000005",
  "materialTextDisabledShadow": "#000003",
  "materialTextInvert": "#00001b",
  "name": "theme",
  "progress": "#00001a",
  "scale": 2,
  "shadow": true,
  "tooltip": "#00001e",
}
`;

exports[`mapFromWindowsTheme should support gradients and shadows and scale 1`] = `
Object {
  "anchor": "#00001c",
  "anchorVisited": "#00001c",
  "borderDark": "#000005",
  "borderDarkest": "#000001",
  "borderLight": "#000004",
  "borderLightest": "#000003",
  "canvas": "#00000c",
  "canvasText": "#00000e",
  "canvasTextDisabled": "#000005",
  "canvasTextDisabledShadow": "#000003",
  "canvasTextInvert": "#00001b",
  "checkmark": "#00000e",
  "checkmarkDisabled": "#000019",
  "desktopBackground": "#000009",
  "flatDark": "#000005",
  "flatLight": "#000004",
  "focusSecondary": "#000003",
  "headerBackground": "linear-gradient(to right, #00000f, #000010)",
  "headerNotActiveBackground": "linear-gradient(to right, #000012, #000011)",
  "headerNotActiveText": "#000013",
  "headerText": "#000014",
  "hoverBackground": "#00001a",
  "material": "#000002",
  "materialDark": "#000012",
  "materialText": "#000006",
  "materialTextDisabled": "#000005",
  "materialTextDisabledShadow": "#000003",
  "materialTextInvert": "#00001b",
  "name": "theme",
  "progress": "#00001a",
  "scale": 1.5,
  "shadow": false,
  "tooltip": "#00001e",
}
`;
